README.md
# Clean Code Times

[![codecov](https://img.shields.io/codecov/c/github/josegoval/clean-code-milliseconds?style=for-the-badge)](https://codecov.io/gh/josegoval/clean-code-milliseconds)
![npm bundle size](https://img.shields.io/bundlephobia/min/clean-code-milliseconds?style=for-the-badge)
[![https://nodei.co/npm/clean-code-milliseconds.png?downloads=true&downloadRank=true&stars=true](https://nodei.co/npm/clean-code-milliseconds.png?downloads=true&downloadRank=true&stars=true)](https://www.npmjs.com/package/clean-code-milliseconds)

`clean-code-milliseconds` is a JS and TS library that help you write clean code milliseconds.

**Feel like supporting this free plugin?**

<a href="https://www.buymeacoffee.com/josegoval" target="_blank"><img src="https://cdn.buymeacoffee.com/buttons/v2/default-yellow.png" alt="Buy Me A Coffee" style="height: 60px !important;width: 217px !important;" ></a>

Are you bored of complex mathematics to figure out what a person wants to express in milliseconds? Make it easy with pre-built constants and converters.

**Table of Contents**

- [Installation](#installation)
- [Example usage](#example-usage)
- [Functions](#functions)
- [Constants](#constants)

## Installation

```bash
npm install --save clean-code-milliseconds
```

or

```bash
yarn add clean-code-milliseconds
```

## Example usage

Import and use as follow with ES6 syntax:

```ts
import { getDaysInMS, FIVE_DAYS_IN_MS } from "clean-code-milliseconds";

console.log(getDaysInMs(5)) 
console.log(FIVE_DAYS_IN_MS) 
```

For node environments without ES6 use `require` as follows:

```ts

const { getDaysInMS, FIVE_DAYS_IN_MS } = require("clean-code-milliseconds")

console.log(getDaysInMs(5)) 
console.log(FIVE_DAYS_IN_MS) 
```

## Functions

- `getSecondsInMs`
- `getMinutesInMs`
- `getHoursInMs`
- `getDaysInMs`
- `getWeeksInMs`
- `getYearsInMs`: By default it counts 365 days, however, if you want to add a leap year add `{ isLeapYear: true }` to the second argument.

## Constants

Let intellisense guide you <(￣︶￣)>

**Seconds**
- `A_SECOND_IN_MS`
- `TWO_SECONDS_IN_MS`
- `THREE_SECONDS_IN_MS`
- `FOUR_SECONDS_IN_MS`
- `FIVE_SECONDS_IN_MS`
- `SIX_SECONDS_IN_MS`
- `SEVEN_SECONDS_IN_MS`
- `EIGHT_SECONDS_IN_MS`
- `NINE_SECONDS_IN_MS`
- `TEN_SECONDS_IN_MS`
- `THIRTY_SECONDS_IN_MS`

**Minutes**
- `A_MINUTE_IN_MS`
- `TWO_MINUTES_IN_MS`
- `THREE_MINUTES_IN_MS`
- `FOUR_MINUTES_IN_MS`
- `FIVE_MINUTES_IN_MS`
- `SIX_MINUTES_IN_MS`
- `SEVEN_MINUTES_IN_MS`
- `EIGHT_MINUTES_IN_MS`
- `NINE_MINUTES_IN_MS`
- `TEN_MINUTES_IN_MS`
- `FIFTEEN_MINUTES_IN_MS`
- `THIRTY_MINUTES_IN_MS`
- `FORTY_FIVE_MINUTES_IN_MS`

**Hours**
- `A_HOUR_IN_MS`
- `TWO_HOURS_IN_MS`
- `THREE_HOURS_IN_MS`
- `FOUR_HOURS_IN_MS`
- `FIVE_HOURS_IN_MS`
- `SIX_HOURS_IN_MS`
- `SEVEN_HOURS_IN_MS`
- `EIGHT_HOURS_IN_MS`
- `NINE_HOURS_IN_MS`
- `TEN_HOURS_IN_MS`

**Days**
- `A_DAY_IN_MS`
- `TWO_DAYS_IN_MS`
- `THREE_DAYS_IN_MS`
- `FOUR_DAYS_IN_MS`
- `FIVE_DAYS_IN_MS`
- `SIX_DAYS_IN_MS`
- `SEVEN_DAYS_IN_MS`
- `EIGHT_DAYS_IN_MS`
- `NINE_DAYS_IN_MS`
- `TEN_DAYS_IN_MS`

**Weeks**
- `A_WEEK_IN_MS`
- `TWO_WEEK_IN_MS`
- `THREE_WEEK_IN_MS`

**Years**
- `A_YEAR_IN_MS`
- `TWO_YEARS_IN_MS`
- `THREE_YEARS_IN_MS`
- `FOUR_YEARS_IN_MS`
- `FIVE_YEARS_IN_MS`
- `SIX_YEARS_IN_MS`
- `SEVEN_YEARS_IN_MS`
- `EIGHT_YEARS_IN_MS`
- `NINE_YEARS_IN_MS`
- `TEN_YEARS_IN_MS`




CHANGELOG.md
## [1.0.1](https://github.com/josegoval/clean-code-milliseconds/compare/v1.0.0...v1.0.1) (2023-09-10)


### Bug Fixes

* docs updated ([f87a525](https://github.com/josegoval/clean-code-milliseconds/commit/f87a5250422278a0e5f46d18a478207352c77567))

# 1.0.0 (2023-09-10)


### Features

* millisecond constants and converters ([d57c936](https://github.com/josegoval/clean-code-milliseconds/commit/d57c9361cac563e2538e6e85b69d5253b2a37314))


src/index.test.ts
it("test", () => {})

src/milliseconds/constants.test.ts
import { A_DAY_IN_MS, A_MINUTE_IN_MS, A_SECOND_IN_MS, A_WEEK_IN_MS, A_YEAR_IN_MS, EIGHT_DAYS_IN_MS, EIGHT_MINUTES_IN_MS, EIGHT_SECONDS_IN_MS, EIGHT_YEARS_IN_MS, FIFTEEN_MINUTES_IN_MS, FIVE_DAYS_IN_MS, FIVE_MINUTES_IN_MS, FIVE_SECONDS_IN_MS, FIVE_YEARS_IN_MS, FORTY_FIVE_MINUTES_IN_MS, FOUR_DAYS_IN_MS, FOUR_MINUTES_IN_MS, FOUR_SECONDS_IN_MS, FOUR_YEARS_IN_MS, NINE_DAYS_IN_MS, NINE_MINUTES_IN_MS, NINE_SECONDS_IN_MS, NINE_YEARS_IN_MS, SEVEN_DAYS_IN_MS, SEVEN_MINUTES_IN_MS, SEVEN_SECONDS_IN_MS, SEVEN_YEARS_IN_MS, SIX_DAYS_IN_MS, SIX_MINUTES_IN_MS, SIX_SECONDS_IN_MS, SIX_YEARS_IN_MS, TEN_DAYS_IN_MS, TEN_MINUTES_IN_MS, TEN_SECONDS_IN_MS, TEN_YEARS_IN_MS, THIRTY_MINUTES_IN_MS, THIRTY_SECONDS_IN_MS, THREE_DAYS_IN_MS, THREE_MINUTES_IN_MS, THREE_SECONDS_IN_MS, THREE_WEEK_IN_MS, THREE_YEARS_IN_MS, TWO_DAYS_IN_MS, TWO_MINUTES_IN_MS, TWO_SECONDS_IN_MS, TWO_WEEK_IN_MS, TWO_YEARS_IN_MS } from "./constants";
import { getDaysInMs, getHoursInMs } from "./converters";

describe("Millisecond module", () => {
	it("converts seconds", ()=> {
		const aSecondInMS = 1000;
		expect(A_SECOND_IN_MS).toBe(aSecondInMS);
		expect(TWO_SECONDS_IN_MS).toBe(aSecondInMS * 2);
		expect(THREE_SECONDS_IN_MS).toBe(aSecondInMS * 3);
		expect(FOUR_SECONDS_IN_MS).toBe(aSecondInMS * 4);
		expect(FIVE_SECONDS_IN_MS).toBe(aSecondInMS * 5);
		expect(SIX_SECONDS_IN_MS).toBe(aSecondInMS * 6);
		expect(SEVEN_SECONDS_IN_MS).toBe(aSecondInMS * 7);
		expect(EIGHT_SECONDS_IN_MS).toBe(aSecondInMS * 8);
		expect(NINE_SECONDS_IN_MS).toBe(aSecondInMS * 9);
		expect(TEN_SECONDS_IN_MS).toBe(aSecondInMS * 10);
		expect(THIRTY_SECONDS_IN_MS).toBe(aSecondInMS * 30);
	});

	it("converts minutes", ()=> {
		expect(A_MINUTE_IN_MS).toBe(A_MINUTE_IN_MS);
		expect(TWO_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 2);
		expect(THREE_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 3);
		expect(FOUR_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 4);
		expect(FIVE_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 5);
		expect(SIX_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 6);
		expect(SEVEN_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 7);
		expect(EIGHT_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 8);
		expect(NINE_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 9);
		expect(TEN_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 10);
		expect(FIFTEEN_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 15);
		expect(THIRTY_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 30);
		expect(FORTY_FIVE_MINUTES_IN_MS).toBe(A_MINUTE_IN_MS * 45);
	});

	it("converts days", ()=> {
		const aDayInMs = getHoursInMs(24);
		expect(A_DAY_IN_MS).toBe(aDayInMs);
		expect(TWO_DAYS_IN_MS).toBe(aDayInMs * 2);
		expect(THREE_DAYS_IN_MS).toBe(aDayInMs * 3);
		expect(FOUR_DAYS_IN_MS).toBe(aDayInMs * 4);
		expect(FIVE_DAYS_IN_MS).toBe(aDayInMs * 5);
		expect(SIX_DAYS_IN_MS).toBe(aDayInMs * 6);
		expect(SEVEN_DAYS_IN_MS).toBe(aDayInMs * 7);
		expect(EIGHT_DAYS_IN_MS).toBe(aDayInMs * 8);
		expect(NINE_DAYS_IN_MS).toBe(aDayInMs * 9);
		expect(TEN_DAYS_IN_MS).toBe(aDayInMs * 10);
	});

	it("converts weeks", ()=> {
		expect(A_WEEK_IN_MS).toBe(SEVEN_DAYS_IN_MS);
		expect(TWO_WEEK_IN_MS).toBe(SEVEN_DAYS_IN_MS * 2);
		expect(THREE_WEEK_IN_MS).toBe(SEVEN_DAYS_IN_MS * 3);
	});

	it("converts years", ()=> {
		const aYearInMs = getDaysInMs(365);
		expect(A_YEAR_IN_MS).toBe(aYearInMs);
		expect(TWO_YEARS_IN_MS).toBe(aYearInMs * 2);
		expect(THREE_YEARS_IN_MS).toBe(aYearInMs * 3);
		expect(FOUR_YEARS_IN_MS).toBe(aYearInMs * 4);
		expect(FIVE_YEARS_IN_MS).toBe(aYearInMs * 5);
		expect(SIX_YEARS_IN_MS).toBe(aYearInMs * 6);
		expect(SEVEN_YEARS_IN_MS).toBe(aYearInMs * 7);
		expect(EIGHT_YEARS_IN_MS).toBe(aYearInMs * 8);
		expect(NINE_YEARS_IN_MS).toBe(aYearInMs * 9);
		expect(TEN_YEARS_IN_MS).toBe(aYearInMs * 10);
	});
});

src/milliseconds/converters.test.ts
import { getDaysInMs, getYearsInMs } from "./converters";

describe("Converters module", () => {
	it("converts leap years", ()=> {
		expect(getYearsInMs(3, { isLeapYear: true })).toBe(getDaysInMs(366) * 3);
	});

});

.eslintrc.js
module.exports = {
	"env": {
		"browser": true,
		"es2021": true,
		"node": true
	},
	"extends": [
		"eslint:recommended",
		"plugin:@typescript-eslint/recommended"
	],
	"overrides": [
		{
			"env": {
				"node": true
			},
			"files": [
				".eslintrc.{js,cjs}"
			],
			"parserOptions": {
				"sourceType": "script"
			}
		}
	],
	"parser": "@typescript-eslint/parser",
	"parserOptions": {
		"ecmaVersion": "latest",
		"sourceType": "module"
	},
	"plugins": [
		"@typescript-eslint",
		"unused-imports"
	],
	"rules": {
		"indent": [
			"error",
			"tab"
		],
		"linebreak-style": [
			"error",
			"windows"
		],
		"quotes": [
			"error",
			"double"
		],
		"semi": [
			"error",
			"always"
		],
		"no-multiple-empty-lines": [
			"error",
			{ max: 1 }
		],
		"space-in-parens": "error",
		"object-curly-spacing": ["error", "always"],
		"space-infix-ops": "error",
		"unused-imports/no-unused-imports": "error",
		"no-duplicate-imports": "error",
	}
};


src/milliseconds/constants.ts
import { getDaysInMs, getHoursInMs, getMinutesInMs, getSecondsInMs, getWeeksInMs, getYearsInMs } from "./converters";

export const A_SECOND_IN_MS = getSecondsInMs(1);
export const TWO_SECONDS_IN_MS = getSecondsInMs(2);
export const THREE_SECONDS_IN_MS = getSecondsInMs(3);
export const FOUR_SECONDS_IN_MS = getSecondsInMs(4);
export const FIVE_SECONDS_IN_MS = getSecondsInMs(5);
export const SIX_SECONDS_IN_MS = getSecondsInMs(6);
export const SEVEN_SECONDS_IN_MS = getSecondsInMs(7);
export const EIGHT_SECONDS_IN_MS = getSecondsInMs(8);
export const NINE_SECONDS_IN_MS = getSecondsInMs(9);
export const TEN_SECONDS_IN_MS = getSecondsInMs(10);
export const THIRTY_SECONDS_IN_MS = getSecondsInMs(30);

export const A_MINUTE_IN_MS = getMinutesInMs(1);
export const TWO_MINUTES_IN_MS = getMinutesInMs(2);
export const THREE_MINUTES_IN_MS = getMinutesInMs(3);
export const FOUR_MINUTES_IN_MS = getMinutesInMs(4);
export const FIVE_MINUTES_IN_MS = getMinutesInMs(5);
export const SIX_MINUTES_IN_MS = getMinutesInMs(6);
export const SEVEN_MINUTES_IN_MS = getMinutesInMs(7);
export const EIGHT_MINUTES_IN_MS = getMinutesInMs(8);
export const NINE_MINUTES_IN_MS = getMinutesInMs(9);
export const TEN_MINUTES_IN_MS = getMinutesInMs(10);
export const FIFTEEN_MINUTES_IN_MS = getMinutesInMs(15);
export const THIRTY_MINUTES_IN_MS = getMinutesInMs(30);
export const FORTY_FIVE_MINUTES_IN_MS = getMinutesInMs(45);

export const A_HOUR_IN_MS = getHoursInMs(1);
export const TWO_HOURS_IN_MS = getHoursInMs(2);
export const THREE_HOURS_IN_MS = getHoursInMs(3);
export const FOUR_HOURS_IN_MS = getHoursInMs(4);
export const FIVE_HOURS_IN_MS = getHoursInMs(5);
export const SIX_HOURS_IN_MS = getHoursInMs(6);
export const SEVEN_HOURS_IN_MS = getHoursInMs(7);
export const EIGHT_HOURS_IN_MS = getHoursInMs(8);
export const NINE_HOURS_IN_MS = getHoursInMs(9);
export const TEN_HOURS_IN_MS = getHoursInMs(10);

export const A_DAY_IN_MS = getDaysInMs(1);
export const TWO_DAYS_IN_MS = getDaysInMs(2);
export const THREE_DAYS_IN_MS = getDaysInMs(3);
export const FOUR_DAYS_IN_MS = getDaysInMs(4);
export const FIVE_DAYS_IN_MS = getDaysInMs(5);
export const SIX_DAYS_IN_MS = getDaysInMs(6);
export const SEVEN_DAYS_IN_MS = getDaysInMs(7);
export const EIGHT_DAYS_IN_MS = getDaysInMs(8);
export const NINE_DAYS_IN_MS = getDaysInMs(9);
export const TEN_DAYS_IN_MS = getDaysInMs(10);

export const A_WEEK_IN_MS = getWeeksInMs(1);
export const TWO_WEEK_IN_MS = getWeeksInMs(2);
export const THREE_WEEK_IN_MS = getWeeksInMs(3);

export const A_YEAR_IN_MS = getYearsInMs(1);
export const TWO_YEARS_IN_MS = getYearsInMs(2);
export const THREE_YEARS_IN_MS = getYearsInMs(3);
export const FOUR_YEARS_IN_MS = getYearsInMs(4);
export const FIVE_YEARS_IN_MS = getYearsInMs(5);
export const SIX_YEARS_IN_MS = getYearsInMs(6);
export const SEVEN_YEARS_IN_MS = getYearsInMs(7);
export const EIGHT_YEARS_IN_MS = getYearsInMs(8);
export const NINE_YEARS_IN_MS = getYearsInMs(9);
export const TEN_YEARS_IN_MS = getYearsInMs(10);
